[ODS-5212]Migration Support for ODS / API 5.2

semalaiappan opened the pull request.

semalaiappan committed:
added script V51 to V52

semalaiappan committed:
Migration support for Sql server 5.2 Added

semalaiappan committed:
Migration support for Postgres 5.2 Added

semalaiappan committed:
Added ValidateJournalEntries Test method  for Postgres 5.2 Added

semalaiappan committed:
Updated Test method for Postgres 5.2

semalaiappan committed:
Scripts updated for Postgres 5.2

semalaiappan committed:
Scripts updated for Postgres 5.2

semalaiappan committed:
Scripts updated for Postgres 5.2

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/MsSql/02Upgrade/v51_to_v52_RecordOwnership/01 Structure/1000-AddOwnershipColumnsForNewTablesV52.sql:
```diff
@@ -0,0 +1,8 @@
+
+ALTER TABLE [edfi].[AssessmentScoreRangeLearningStandard] ADD [CreatedByOwnershipTokenId] SMALLINT NULL;
+ALTER TABLE [edfi].[StudentDisciplineIncidentBehaviorAssociation] ADD [CreatedByOwnershipTokenId] SMALLINT NULL;
+ALTER TABLE [edfi].[StudentDisciplineIncidentNonOffenderAssociation] ADD [CreatedByOwnershipTokenId] SMALLINT NULL;
```
Comment is actually for the rename operation below. Why are we renaming several files to v53 in a branch for API 5.2?

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql:
```diff
@@ -0,0 +1,230 @@
+IS 'A reference to the behavior(s) by the student that led or contributed to this specific action.';
+
+COMMENT ON COLUMN edfi.DisciplineActionStudentDisciplineIncidentBehaviorAssociation.BehaviorDescriptorId
+IS 'Describes behavior by catery.';
```
Typo: `catery` should be `category`.

semalaiappan replied:
fixed it

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql:
```diff
@@ -0,0 +1,230 @@
+COMMENT ON TABLE edfi.StudentDisciplineIncidentBehaviorAssociation IS 'This association describes the behavior of students involved in a discipline incident.';
+
+
+COMMENT ON COLUMN edfi.StudentDisciplineIncidentBehaviorAssociationDisciplineIn_ae6a21.BehaviorDescriptorId IS 'Describes behavior by catery.';
```
Typo: `catery` should be `category`.

semalaiappan replied:
fixed it

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1040-AddExtendedPropertiesForV52.sql:
```diff
@@ -0,0 +1,230 @@
+COMMENT ON TABLE edfi.StaffAncestryEthnicOrigin IS 'The original peoples or cultures with which the individual identifies.' ;
+
+
+COMMENT ON COLUMN edfi.StudentDisciplineIncidentBehaviorAssociation.BehaviorDescriptorId IS 'Describes behavior by catery.';
```
Typo: `catery` should be `category`.

semalaiappan replied:
fixed it

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql:
```diff
@@ -0,0 +1,245 @@
+    Discriminator                         VARCHAR (128)   NULL,
+    CreateDate                            TIMESTAMP   DEFAULT current_timestamp NOT NULL ,
+    LastModifiedDate                      TIMESTAMP   DEFAULT current_timestamp NOT NULL ,
+    Id                                    UUID  DEFAULT gen_random_uuid() NOT NULL,
```
In ODS for v5.2 from the package, this line seems like it should end as `UUID  DEFAULT public.gen_random_uuid() NOT NULL,`

semalaiappan replied:
fixed it

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql:
```diff
@@ -0,0 +1,245 @@
+    Discriminator               VARCHAR (128)   NULL,
+    CreateDate                  TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    LastModifiedDate            TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    Id                          UUID DEFAULT gen_random_uuid() NOT NULL,
```
In ODS for v5.2 from the package, this line seems like it should end as `UUID  DEFAULT public.gen_random_uuid() NOT NULL,`

semalaiappan replied:
fixed it

gmcelhanon commented on EdFi.Ods.Utilities.Migration/Scripts/PgSql/02Upgrade/v51_to_v52/01 Structure/1010-AddNewTablesForV52.sql:
```diff
@@ -0,0 +1,245 @@
+    Discriminator      VARCHAR (128)   NULL,
+    CreateDate         TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    LastModifiedDate   TIMESTAMP    DEFAULT current_timestamp NOT NULL,
+    Id                 UUID  DEFAULT gen_random_uuid() NOT NULL,
```
In ODS for v5.2 from the package, this line seems like it should end as `UUID  DEFAULT public.gen_random_uuid() NOT NULL,`

semalaiappan replied:
fixed it

gmcelhanon reviewed (changes requested):
Both migrations worked without error. There were some minor (probably whitespace) differences on an extended property (on `edfi.StudentDisciplineIncidentAssociation` table in SQL Server and PostgreSQL). Additionally, there were a few occurrences of a very minor discrepancy on the definition of the `Id` column on some tables, and there was a typo in the comment for the `BehaviorDescriptorId` column in 3 places.

semalaiappan committed:
review comments fix

semalaiappan committed:
Added ExtendedProperty for studentdisciplineincidentassociation

semalaiappan committed:
trailing space fix in Extended Property

gmcelhanon committed:
Remove trailing space on column comment in PostgreSQL

`edfi.studentdisciplineincidentassociation.studentparticipationcodedescriptorid`
